lesson-04-memory-management/mm/xxx_km.c

```diff
@@ -32,13 +33,18 @@ int __init x_init(void) {
    x_class = class_create( THIS_MODULE, "x-class" );
    if( IS_ERR( x_class ) ) printk( "bad class create\n" );
    res = class_create_file( x_class, &class_attr_xxx );
+
+   buf_msg = (char*) kmalloc(LEN_MSG, 0);
```

Comment:
Dynamic allocation means doing that when you need to store data.
You shouldn't allocate buffer of predefined LEN_MSG size, but rather use actual required buffer size.